### PR TITLE
Fix reprovisioning of VM when we attach and detach data disk.

### PR DESCRIPTION
**Pull request by VaijanathB**
This is the same pull request as Fixes issue when we need to attach/detach data disks #778 . Creating this as I did not wanted to reapply the 42 commits.

### CODE REVIEW

**Comment by VaijanathB** [comment]
This is already reviewed and tested as #778

**Comment by VaijanathB** [comment]
Run the tests with test fix and no new failures. @tombuildsstuff Can you please merge this ?

**Comment by VaijanathB** [comment]
Sapient verified that this Pull Request solves their problem.

File: azurerm/resource_arm_virtual_machine_managed_disks_test.go [hunk 1]
```diff
@@ -227,7 +227,7 @@ func TestAccAzureRMVirtualMachine_bugAzureRM33(t *testing.T) {
 	})
 }
 
-func TestAccAzureRMVirtualMachine_changeStorageDataDiskCreationOption(t *testing.T) {
+func TestAccAzureRMVirtualMachine_attachSecondDataDiskWithAttachOption(t *testing.T) {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_virtual_machine_managed_disks_test.go]: This test currently fails:

```
------- Stdout: -------
=== RUN   TestAccAzureRMVirtualMachine_attachSecondDataDiskWithAttachOption
--- FAIL: TestAccAzureRMVirtualMachine_attachSecondDataDiskWithAttachOption (706.71s)
    testing.go:459: Step 1 error: Check failed: Check 3/3 error: azurerm_virtual_machine.test: Attribute 'storage_data_disk.0.create_option' expected "Attach", got "Empty"
FAIL
```

I've pushed a commit to fix this

Comment (tombuildsstuff) [inline azurerm/resource_arm_virtual_machine_managed_disks_test.go]: This test currently fails:

```
------- Stdout: -------
=== RUN   TestAccAzureRMVirtualMachine_attachSecondDataDiskWithAttachOption
--- FAIL: TestAccAzureRMVirtualMachine_attachSecondDataDiskWithAttachOption (706.71s)
    testing.go:459: Step 1 error: Check failed: Check 3/3 error: azurerm_virtual_machine.test: Attribute 'storage_data_disk.0.create_option' expected "Attach", got "Empty"
FAIL
```

I've pushed a commit to fix this

**Comment by tombuildsstuff** [comment]
Ignoring an Azure provisioning failure - the tests pass:

screen shot 2018-02-14 at 17 08 29

**Pull request merged**